### ISSUE TITLE
Output files: prepend `CircuitName_` (or something else)

### ISSUE THREAD

**Issue by PMeira** [dss-extensions/dss_capi#94]
Some files already do use `CircuitName_` but not all of them. For consistency and for usage in parallel scenarios, it's a good idea to prepend something to differentiate outputs.

**Comment by PMeira** [comment]
Adding this as an option could be less annoying and more compatible with older code. Moving to 0.12...

**Comment by PMeira** [comment]
Reviewing most of the file output, it seems most of the outputs already use `CircuitName_`. 
For the PM functions, `_Name` (empty normally or index of the actor in PM mode) is also used for monitors and energy meters.

`CircuitName_` is the same as the `CaseName` + `_` (`CaseName` could be removed to avoid redundancy).

From the help string:
> Name of case for yearly simulations with demand interval data. Becomes the name of the subdirectory under which all the year data are stored. Default = circuit name
>
> Side Effect: Sets the prefix for output files

I think that's enough to avoid issues.

Some output files such as DebugTraces and other occasional outputs, shapes in general, circuit dumps, etc. don't use any prefix/suffix, but that's seems reasonable and I don't think it makes sense to run large-scale parallel simulations with those.

Another aspect is `OutputDirectory`. This one is now used for every thing file instead of the mixed `CurrentDirectory` (= current working directory of the process) and `DataPath`. The new behavior is already documented in the changelog. `DataPath` is exposed in the APIs.

So, in conclusion, unless we missed something, the users have a good level of control of where the files will go and their prefixes/infixes through `DSS.DataPath`, the circuit's name, and the `CaseName` option.
